Source/Core/Core/API/Events.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <vector>
+#include "Common/CommonTypes.h"
+
+namespace API
+{
+
+namespace Events
+{
+  // Each event also needs a manual template instantiation in Events.cpp
+  struct Frameadvance {};
+  struct Memory { bool write; u32 addr; u32 value; };
+  struct Interrupt { u32 cause_mask; bool set; };
+
+}  // namespace API::Events
+
+// a listener on T is any function that takes a const reference to T as argument
+template <typename T>
+using Listener = void (*)(const T&);
```

Comment:
Sure. I store the functions in a list though, and need to be able to remove them later on again. Apparently std::function doesn't implement `==` properly, and I haven't found out how I would implement unregistering listeners otherwise. Any ideas?